addon/features.js

```diff
@@ -1,4 +1,11 @@
 export default {
+  flags: {},
+  setup: function(flags) {
+    this.flags = flags;
+  },
+  set: function(flag, enabled) {
```

Comment:
I know I used set in my pseudo code on one of the issues, but I think set will be confusing and pose problems if we convert this to an ember object. Maybe setFeature? I don't really have a good naming idea, unfortunately.